### PR TITLE
cleanup and changes to work on dotnet for linux

### PR DESCRIPTION
**Pull request by danyelf**
Updates based on conversations with @cartermp  and @vreynolds 

## Which problem is this PR solving?

- docs doesn't mention run tilt from dotnet directory
- docs doesn't list all necessary envars
- `docker compose` doesn't work on linux
- `docker COPY` [requires `/` for directories on linux](https://stackoverflow.com/questions/53650492/when-using-copy-with-more-than-one-source-file-the-destination-must-be-a-direct/53715989)
- settings files still refer to api-dogfood
- The tilt entry `docker compose exec` fails with "not a TTY", so adding `-t` makes it a TTY.

## Short description of the changes

-

### CODE REVIEW

File: dotnet/frontend/appsettings.json [hunk 1]
```diff
@@ -8,9 +8,6 @@
   },
   "AllowedHosts": "*",
   "Honeycomb": {
-    "ServiceName": "frontend-dotnet",
-    "Endpoint": "https://api-dogfood.honeycomb.io",
-    "ApiKey": "",
-    "Dataset": "greetings"
```
Comment (cartermp) [inline dotnet/frontend/appsettings.json:14]: Losing `Dataset` does seem a little strange. This definitely gets into "we're in tilt/docker territory" and diverges from our docs.

Is that fine?

Comment (danyelf) [reply dotnet/frontend/appsettings.json:14]: The dataset should not be set for each Service  -- it's the other global that you need to set.

To counter your question -- what should happen if `HONEYCOMB_DATASET` and  this `Dataset` disagree?

Comment (danyelf) [reply dotnet/frontend/appsettings.json:14]: (Noting of course that this is all pre-E&S transition)

Comment (cartermp) [reply dotnet/frontend/appsettings.json:14]: >To counter your question -- what should happen if HONEYCOMB_DATASET and this Dataset disagree?

In this case, the environment variable will win.

Sounds good overall!

**Review by cartermp — APPROVED** [review]

File: Tiltfile [hunk 2]
```diff
@@ -183,31 +183,31 @@ def launch_node_year_service(auto_init=True):
   launch_node_svc("year-node", dirname="node/year-service", auto_init=auto_init)
 
 # Launch one of each of these types of services. Go services init by default
-launch_go_frontend()
+# launch_go_frontend()
```
Comment (vreynolds) [inline Tiltfile:186]: We usually don't commit the launch diffs, and keep it defaulting to Go

File: dotnet/frontend/appsettings.json (hunk 1 above)
Comment (vreynolds) [reply dotnet/frontend/appsettings.json:14]: I see your point Phillip. I think we should either:
- delete the entire honeycomb/otlp section from the configs and embrace that we're in tilt world
- keep the config (with all the values) as an example, understanding that env vars override it

**Review by vreynolds — CHANGES_REQUESTED** [review]
Dotnet still works on my mac! One question about appsettings.json and need to roll back the launch switcheroo.

File: dotnet/frontend/appsettings.json (hunk 1 above)
Comment (JamieDanielson) [reply dotnet/frontend/appsettings.json:14]: I'm a big fan of keeping this all runnable and understandable without Tilt, even if it means some env vars are just commented out. We intend to keep using this as an up-to-date example repo to point people to for multiple services, mixed instrumentations, etc... so it's probably best to ensure it's very clear where/how these variables can be set.

Comment (MikeGoldsmith) [reply dotnet/frontend/appsettings.json:14]: I agree with @JamieDanielson, someone just want to refer to our example as code example and having all the settings 'hidden' in the tilt file can be confusing.

File: README.md [hunk 3]
```diff
@@ -28,6 +28,7 @@ variables:
 
 - HONEYCOMB_API_KEY - Your honeycomb API key
 - HONEYCOMB_DATASET - The name of the dataset you want to write to
+- $OTEL_EXPORTER_OTLP_ENDPOINT=`https://api.honeycomb.io`
```
Comment (MikeGoldsmith) [inline README.md:31]: You can set `SERVICE_NAME` env var to set service name.

File: dotnet/frontend/appsettings.json (hunk 1 above)
Comment (danyelf) [reply dotnet/frontend/appsettings.json:14]: got it. Adding them back in. 

```json
    "ServiceName": "frontend-dotnet",
    "Endpoint": "https://api.honeycomb.io",
    "ApiKey": "",
    "Dataset": "greetings"
```

File: Tiltfile (hunk 2 above)
Comment (danyelf) [reply Tiltfile:186]: Ooops! Sorry about that.

**Comment by danyelf** [comment]
Addressed issues

File: README.md [hunk 4]
```diff
@@ -26,13 +26,18 @@ OpenTelemetry.
...
 Some services use vanilla OTEL:
-- OTEL_EXPORTER_OTLP_HEADERS='x-honeycomb-team=api-key,x-honeycomb-dataset=greetings'
+
+- `OTEL_EXPORTER_OTLP_HEADERS='x-honeycomb-team=api-key,x-honeycomb-dataset=greetings'`
+
 If using dogfood:
-- OTEL_EXPORTER_OTLP_ENDPOINT=https://api-dogfood.honeycomb.io
-- HONEYCOMB_API_ENDPOINT=https://api-dogfood.honeycomb.io
+
+- `OTEL_EXPORTER_OTLP_ENDPOINT=https://api-dogfood.honeycomb.io`
```
Comment (danyelf) [inline README.md:39]: I formatted these -- but this is a public-facing document. Do we want `dogfood` mentioned in it at ALL?

File: README.md (hunk 3 above)
Comment (danyelf) [reply README.md]: would you like us to add that too? we didn't because each service has its own name

File: README.md (hunk 4 above)
Comment (vreynolds) [reply README.md:39]: I can generalize it in another PR

**Review by vreynolds — APPROVED** [review]
🌮 🎉

**Pull request merged**